app/views/devise/invitations/new.html.haml

```diff
@@ -9,44 +9,47 @@
       %div.column-full
         = f.text_field field
 
-  - if current_user.admin?
-    .form-group
-      = f.hidden_field :role, value: params[:role]
-      - if params[:role] == 'custodian'
-        = f.label :registers, class: "form-label"
-        = f.nested_fields_for :teams, f.object.teams.new do |team|
-          %div.grid-row.form-group
-            %div.column-one-half
-              = team.select :registers, beta_registers_by_name.collect{ |r| [prepare_register_name(r), r]}, {include_blank: "", multiple: true }, { class: "form-control autocomplete" }
-            %div.column-one-half
-              = team.remove_nested_fields_link 'Remove register'
-
-        = f.add_nested_fields_link :teams, 'Add register'
-
+  - if params[:role] == 'admin'
```

Comment:
since the admin tag is moved to the user level it means that it not longer suits the model. 

The model is inviting team_members and if they don't exist then generate a user. Its causing this horrible embedded if that has the code smell that it doesn't fit in the model.

It might be worth a separate invite for admin users.